bound/kt/src/test/kotlin/web5/sdk/crypto/verifiers/Ed25519VerifierTest.kt

```diff
@@ -1,41 +1,127 @@
 package web5.sdk.crypto.verifiers
 
-import org.junit.jupiter.api.Test
-import org.junit.jupiter.api.Assertions.assertTrue
-import org.junit.jupiter.api.Assertions.assertFalse
+import org.junit.jupiter.api.*
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.fail
+import web5.sdk.UnitTestSuite
+import web5.sdk.crypto.Ed25519Generator
 import web5.sdk.crypto.keys.Jwk
 import web5.sdk.crypto.signers.Ed25519Signer
-import web5.sdk.rust.ed25519GeneratorGenerate as rustCoreEd25519GeneratorGenerate
+import web5.sdk.rust.Web5Exception
 
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
 class Ed25519VerifierTest {
 
+    private val testSuite = UnitTestSuite("ed25519_verify")
+
+    @AfterAll
+    fun verifyAllTestsIncluded() {
+        if (testSuite.tests.isNotEmpty()) {
+            println("The following tests were not included or executed:")
+            testSuite.tests.forEach { println(it) }
+            fail("Not all tests were executed! ${this.testSuite.tests}")
+        }
+    }
+
+    private fun generateKeys(): Pair<Jwk, Jwk> {
+        val privateJwk = Ed25519Generator.generate()
+        val publicJwk = privateJwk.copy(d = null)
+        return Pair(publicJwk, privateJwk)
+    }
+
+    @Test
+    fun test_with_valid_signature() {
+        this.testSuite.include()
+
+        val (publicJwk, privateJwk) = generateKeys()
+        val signer = Ed25519Signer(privateJwk)
+        val verifier = Ed25519Verifier(publicJwk)
+
+        val message = "Test message".toByteArray()
+        val signature = signer.sign(message)
+
+        val verifyResult = runCatching { verifier.verify(message, signature) }
+
+        assertTrue(verifyResult.isSuccess, "Verification should succeed with a valid signature")
+    }
+
+    @Test
+    fun test_with_private_key() {
+        this.testSuite.include()
+
+        val (_, privateJwk) = generateKeys()
+        val verifier = Ed25519Verifier(privateJwk) // this is not allowed
```

Comment:
If this is not allowed, should this line throw?

Reply:
I had the same thought, but we have a test vector ([here](https://github.com/TBD54566975/web5-spec/blob/7c4e7fc5206ee3f50c05383e61be0046623f7dc9/test-vectors/crypto_ed25519/verify.json#L110)) which asserts this behavior, so if we did it in the constructor then we'd be doing it twice. 

@nitro-neal any thoughts on changing the test vector?